libraries/lib-module-manager/PluginManager.cpp

```diff
@@ -389,26 +394,18 @@ void PluginManager::Initialize(FileConfigFactory factory)
 
 void PluginManager::Terminate()
 {
-   // Get rid of all non-module plugins first
-   PluginMap::iterator iter = mPlugins.begin();
-   while (iter != mPlugins.end())
+   // Get rid of all non-module(effects?) plugins first
+   auto iter = mRegisteredPlugins.begin();
+   for(auto& p : mRegisteredPlugins)
    {
-      PluginDescriptor & plug = iter->second;
-      if (plug.GetPluginType() == PluginTypeEffect)
-      {
-         mPlugins.erase(iter++);
-         continue;
-      }
-
-      ++iter;
+      auto& desc = p.second;
+      if(desc.GetPluginType() == PluginTypeEffect)
+         mLoadedInterfaces.erase(desc.GetID());
    }
 
-   // Now get rid of the modules
-   iter = mPlugins.begin();
-   while (iter != mPlugins.end())
-   {
-      mPlugins.erase(iter++);
-   }
+   // Now get rid of others
+   mRegisteredPlugins.clear();
+   mLoadedInterfaces.clear();
```

Comment:
My last remark for this round of review:  I understand these changes of PluginManager and PluginDescriptor but I’m waiting to see how they are used in the next rebasing of the other work for plugin validation.

The suggestion in old comments that PluginDescriptor might become a root of a class hierarchy— this didn’t happen here and I suppose it will not, or else I would have seen it in these preliminaries.  No need yet for that other idea.

Reply:
Alternative might be use attached objects instead of building a hierachy